contract-examples/contracts/ExampleRewardManager.sol

```diff
@@ -31,3 +34,107 @@ contract ExampleRewardManager is Ownable {
     return rewardManager.areFeeRecipientsAllowed();
   }
 }
+
+contract ExampleRewardManagerTest is DSTest {
+  uint blackholeBalance;
+  ExampleRewardManager exampleReceiveFees;
+  uint exampleBalance;
+
+  function setUp() public {
+    blackholeBalance = BLACKHOLE_ADDRESS.balance;
+  }
+
+  function test_sendFeesToBlackhole() public {
+    blackholeBalance = BLACKHOLE_ADDRESS.balance;
+  }
+
+  function test_checkSendFeesToBlackhole() public {
+    assertGt(BLACKHOLE_ADDRESS.balance, blackholeBalance);
+  }
+
+  function test_doesNotSetRewardAddressBeforeEnabled() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    assertEq(rewardManager.readAllowList(exampleAddress), 0);
+
+    try example.setRewardAddress(exampleAddress) {
+      assertTrue(false, "setRewardAddress should fail");
+    } catch {} // TODO should match on an error to make sure that this is failing in the way that's expected
+  }
+
+  function test_setEnabled() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    assertEq(rewardManager.readAllowList(exampleAddress), 0);
+    rewardManager.setEnabled(exampleAddress);
+    assertEq(rewardManager.readAllowList(exampleAddress), 1);
+  }
+
+  function test_setRewardAddress() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    rewardManager.setEnabled(exampleAddress);
+    example.setRewardAddress(exampleAddress);
+
+    assertEq(example.currentRewardAddress(), exampleAddress);
+  }
+
+  function test_setupReceiveFees() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    rewardManager.setEnabled(exampleAddress);
+    example.setRewardAddress(exampleAddress);
+
+    exampleReceiveFees = example;
+    exampleBalance = exampleAddress.balance;
+  }
+
+  function test_receiveFees() public {
+    // noop
```

Comment:
why we have this function?

Reply:
Some tests require multiple transactions. Adding a noop with a specific function name hints at the purpose of the transaction. 

In order to understand the tests, you need to look at how the functions are being called:
https://github.com/ava-labs/subnet-evm/blob/3eec654c97d73dfc7a498e9ad1e19f5ad0bc1a1b/contract-examples/test/reward_manager.ts#L40

`test_setupReceiveFees` -> `test_receiveFees` -> `test_checkReceiveFees` are all called in conjunction.

Reply:
I see. Can we add some comments why this function is empty?

Reply:
done